src/components/AddBook.js

```diff
@@ -1,14 +1,43 @@
-import React from 'react';
+import { React, useState } from 'react';
+import { PropTypes } from 'prop-types';
 
-const AddBook = () => (
-  <div className="form">
-    <h2>Add New Book</h2>
-    <form>
-      <input type="text" name="title" placeholder="Title" />
-      <input type="text" name="author" placeholder="Author" />
-      <button type="submit" id="add-book">ADD BOOK</button>
-    </form>
-  </div>
-);
+const AddBook = ({ AddNewBook }) => {
+  const [NewBook, setNewBook] = useState({ title: '', author: '' });
+
+  return (
+    <div id="form_container">
+      <h2>Add New Book</h2>
+      <form
+        onSubmit={(e) => {
+          AddNewBook(e, NewBook);
+          setNewBook({ title: '', author: '' });
+        }}
+      >
+        <input
+          type="text"
+          name="title"
+          placeholder="Title"
+          value={NewBook.title}
+          required
+          onChange={(e) => setNewBook({ ...NewBook, title: e.target.value })}
+        />
+        <input
+          type="text"
+          name="author"
+          placeholder="Author"
+          value={NewBook.author}
+          required
+          onChange={(e) => setNewBook({ ...NewBook, author: e.target.value })}
+        />
+        <button type="submit" id="add_book">
+          Add Book
+        </button>
```

Comment:
- [ ] As part of the project requirements please make sure to create a custom `<Button/>` component instead of using the default `<button>` tag here and pass the event handler for adding a book as a prop.
- [ ] Kindly take note that you should dispatch this action by importing `useDispatch` and you shouldn't submit the data for each input change like you've done currently.